design/use_cases/collision_avoidance_use_case.md

```diff
@@ -0,0 +1,14 @@
+# Collision Avoidance
+As a Robot user I want to my robot to avoid colliding with people or objects so that it won't damage anything or hurt anyone
+
+## More details
+- Why is this needed?
+   - This is needed for indoor and outdoor robot navigation in most (all?) cases
+   - Example: a robot in a warehouse should avoid colliding into the walls or shelving, and dynamically avoid people that cross its path
+
+- What is the expected user interaction?
+   - The user should be able to walk in front of a robot and it should avoid crashing into that person
+
+- Are there any non-functional requirements? (build system, tools, performance, etc)
```

Comment:
I'm not sure if *collision recovery* should be considered as a use case (what the user expects to do when the robot when crashes into something to put it back in operation) or that comes later as a error handling requirement.

Reply:
@orduno - yes we will need a collision recovery use case. We probably need some for other use cases such as "I'm lost" or "I'm stuck and can't move" or "I have a malfunction", etc.